platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapboxMap.java

```diff
@@ -958,115 +738,31 @@ private void setApiBaseUrl(@NonNull MapboxMapOptions options) {
   // Styling
   //
 
-  /**
-   * <p>
-   * Loads a new map style asynchronous from the specified URL.
-   * </p>
-   * {@code url} can take the following forms:
-   * <ul>
-   * <li>{@code Style.*}: load one of the bundled styles in {@link Style}.</li>
-   * <li>{@code mapbox://styles/<user>/<style>}:
-   * loads the style from a <a href="https://www.mapbox.com/account/">Mapbox account.</a>
-   * {@code user} is your username. {@code style} is the ID of your custom
-   * style created in <a href="https://www.mapbox.com/studio">Mapbox Studio</a>.</li>
-   * <li>{@code http://...} or {@code https://...}:
-   * loads the style over the Internet from any web server.</li>
-   * <li>{@code asset://...}:
-   * loads the style from the APK {@code assets/} directory.
-   * This is used to load a style bundled with your app.</li>
-   * <li>{@code null}: loads the default {@link Style#MAPBOX_STREETS} style.</li>
-   * </ul>
-   * <p>
-   * This method is asynchronous and will return before the style finishes loading.
-   * If you wish to wait for the map to finish loading, listen to the {@link MapView.OnDidFinishLoadingStyleListener}
-   * callback or use the {@link #setStyleUrl(String, OnStyleLoadedListener)} method instead.
-   * </p>
-   * If the style fails to load or an invalid style URL is set, the map view will become blank.
-   * An error message will be logged in the Android logcat and {@link MapView.OnDidFailLoadingMapListener} callback
-   * will be triggered.
-   *
-   * @param url The URL of the map style
-   * @see Style
-   */
-  public void setStyleUrl(@NonNull String url) {
-    setStyleUrl(url, null);
+  public void setStyle(@Style.StyleUrl String style) {
```

Comment:
Such a setup would allow to hide the asynchronous behaviour of style loading. This could work if we would add bookkeeping of all sources/layers added to the style between the style load has finished (which is feasible) but also results in the following (incorrect) interaction:

```java
       Style style = mapboxMap.setStyle(new Style.Builder()
          .withSource(someSource)
          .fromUrl(Style.MAPBOX_STREETS)
        );

        assert style.getSources().size() == 1;
```

Such an interaction was one of the main reasons why I wanted Style.java in the first place. Avoid unexpected results that come with Style/Layer/Sources/etc. Instead of working around asynchronous behaviour, I'm making this clear in the API and logic itself, `MapboxMap#getStyle()` returns null until the style has finished loading. To me this produces the expected behaviour in any use-case. It's up to the developer to adhere to the nullability. They should only call `MapboxMap#getStyle()` if they are certain about the finished style loading event and if they aren't sure they should use `MapboxMap#getStyle(StyleLoadedCallback callback)` instead.

Reply:
This makes perfect sense.